### ISSUE TITLE
Make package run on client and server for server-side rendering

### ISSUE THREAD

**Issue by stubailo** [mrphu3074/react-material-ui#5]
I think all it would take is removing the client restriction from your `package.js`.

**Comment by stubailo** [comment]
You might also need to do something to make the tap event plugin run on the client only, but I'm not sure - maybe it will be OK.

**Comment by mrphu3074** [comment]
I think removing the client restriction is easy. It will make react components available on server side. But i will do that if you have good solution about ReactJs server side rendering .

**Comment by stubailo** [comment]
I think allowing this to run on the server comes first; then people like @arunoda can work on SSR. It's exactly zero cost for the package.

**Comment by mrphu3074** [comment]
@stubailo yeah, i like packages of @arunoda.
let's me try!

**Comment by mrphu3074** [comment]
I tried, we can use flow-router-ssr to render MUI in server side. But the problem need to solve is make sure touchTap event work.

**Comment by ShockiTV** [comment]
I tried it and it seems to work when used this way https://github.com/ShockiTV/flow-ssr-mui-example

**Comment by mrphu3074** [comment]
thanks @ShockiTV , let's me try

**Comment by mrphu3074** [comment]
@ShockiTV its work, but only development mode. i'm investigating it.

**Comment by gdub01** [comment]
I have the same issue with SSR when deploying. It works locally... but trying to run it on whatever.meteor.com makes it crash. 

I did this to package.js to make it run on the client and server:

Package.onUse(function(api){

```
api.use(['react@0.1.3'], ["client", "server"]);
api.use(['cosmos:browserify@0.5.0'], ["client", "server"]);

api.add_files('window.react.js', 'client');

api.addFiles([
    'client.browserify.options.json',
    'client.browserify.js'
], ["client", "server"]);

api.export(["MUI", "injectTapEventPlugin"], ["client", "server"]);
```

});

My logs in production start with:

[Wed Aug 05 2015 23:42:12 GMT+0000 (UTC)] WARNING TypeError: Cannot read
property 'require' of undefined
at Object.<anonymous>
(packages/node_modules/material-ui/lib/app-bar.js:3:1)
at Object../icon-button
(packages/node_modules/material-ui/lib/app-bar.js:220:1)
at s
(packages/../../../../../../.meteor/packages/cosmos_browserify/.0.5.0.1xupaca++os+web.browser+web.cordova/plugin.CosmosBrowserify.os/npm/CosmosBrowserify/node_modules/browserify/node_modules/browser-pack/_prelude.js:1:1)

.... etc.

It appears you need window.react.js', 'client' .. it can't be client and server or it doesn't work locally either.

**Comment by mrphu3074** [comment]
```
api.addFiles([
    'client.browserify.options.json',
    'client.browserify.js'
], ["client", "server"]);

```

we can't use this way. Server side don't know `require`. We must use `Npm.require`. Maybe need to create new file for serverside

**Comment by rozzzly** [comment]
Hey guys read this https://github.com/meteor/react-packages/issues/83 It's not a sever/client issue. It's the new version of the react package. There's a one line fix. fml I spent so long trying to figure out something fixed one line. :sob:

**Comment by mrphu3074** [comment]
thanks @rozzzly . You're right. Maybe need find out react's stable version

**Comment by ShockiTV** [comment]
After merging that rozzzly's PR it seems to run quite OK on meteor test environment http://flow-ssr-mui-example.meteor.com

**Comment by mrphu3074** [comment]
@ShockiTV thank you a lot, i going to finish this package with support ssr.

**Comment by mrphu3074** [comment]
Hi all,
My package supported SSR with @ShockiTV's help.
I created a demo with minimum example.
http://material-ui-ssr.meteor.com/
and source: https://github.com/mrphu3074/react-material-ui-ssr-examples

**Comment by mrphu3074** [comment]
One more thing, about Menu's components

```
  //We're working on migrating some of our components to use a new implementation of menus.
    //If you'd like to use the new menu before our migration is complete, please directly
    //require them like this:

    let MenuItem = require('material-ui/lib/menus/menu-item');
    let MenuDivider = require('material-ui/lib/menus/menu-divider');
```

They working on Menu's components. So i fixed this issue in ver 0.1.5
Please use

```
MUI.Libs.Menu
MUI.Libs.MenuItem
MUI.Libs.MenuDivider
```
